examples/hooks/RecordHooks/conditionallyModifyField.spec.ts

```diff
@@ -0,0 +1,76 @@
+import { Sheet, Workbook, BooleanField, TextField } from '@flatfile/configure'
+import { SheetTester } from '../../../src/utils/testing/SheetTester'
+
+//this Data Hook can be used to update the contents of a cell based on whether or not another cell passes RegEx validation.
+//This method of setting fields at validations at the top of a sheet allows you to set all RegEx and messages in one easily referencable place.
+const FIELDS = {
+  accountId: {
+    regex: /^.{1,20}$/,
+    message: 'Account Number must be 20 characters or less',
+  },
+}
+
+const testSheet = new Sheet(
+  'Test Sheet',
+  {
+    accountId: TextField(),
+    accountStatus: BooleanField(),
+  },
+  {
+    recordCompute: (record) => {
+      //ensuring the accountId is returned as a string
```

Comment:
because accountID is a `TextField` we know it will be `string` or possibly `null`.  Given our existing typing implemenation, we only know that accountID could be one of `null`, `string`, `number`, `date`, `boolean`.  Try this code instead:

```
//make sure we actually have a string, then continue processing
const accountID = record.get('accountID')
if(typeof accountID === 'string') { // this type guard assures typescript that we are dealing with a string
     if (FIELDS['accountId'].regex.test(accountNum) === false) {
        record.set('accountStatus', false)
        record.addError('accountId', FIELDS.accountId.message)
      } else {
        record.set('accountStatus', true)
      }
```

I would recommend explicitly handling the `else` case, where `accountID` is null.

```